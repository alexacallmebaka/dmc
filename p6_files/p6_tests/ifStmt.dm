main : () int  {
  a : int = 10;
  b : bool = true;

  if (b){
    a = 0;
    b = false;
  }
  a = 2;

  if (b) {
    a = 0;
    b = false;
  } else {
    a = 2;
    b = true;
  }

  return a;
}
