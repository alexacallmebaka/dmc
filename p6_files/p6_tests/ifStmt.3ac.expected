[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN main LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
[END main LOCALS]
main:       enter main
            [a] := 10
            [b] := 1
            IFZ [b] GOTO lbl_1
            [a] := 0
            [b] := 0
lbl_1:      nop
            [a] := 2
            IFZ [b] GOTO lbl_2
            [a] := 0
            [b] := 0
            goto lbl_3
lbl_2:      nop
            [a] := 2
            [b] := 1
lbl_3:      nop
            setret [a]
            goto lbl_0
lbl_0:      leave main
